src/main/java/be/seeseemelk/mockbukkit/ServerMock.java

```diff
@@ -1393,29 +1396,30 @@ public ItemStack createExplorerMap(World world, Location location, StructureType
 	@Override
 	public KeyedBossBar createBossBar(NamespacedKey key, String title, BarColor color, BarStyle style, BarFlag... flags)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Validate.notNull(key, "A NamespacedKey must never be null");
+		KeyedBossBarMock bar = new KeyedBossBarMock(key, title, color, style, flags);
+		bossBars.put(key, bar);
+		return bar;
 	}
 
 	@Override
 	public Iterator<KeyedBossBar> getBossBars()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return ((List) new ArrayList<KeyedBossBarMock>(bossBars.values())).iterator();
```

Comment:
This isn't an ideal solution, I'd go for something like this:
```suggestion
		return bossBars.values().stream().map(bossbar -> (KeyedBossBar) bossbar).iterator();
```

I haven't tested it but this should be a bit cleaner as it avoids unsafe type casting and generic types.

Reply:
Ok. Good to know thanks!